### PR TITLE
Add client library for IRM

### PR DESCRIPTION
**Pull request by busunkim96**

### CODE REVIEW

File: irm/setup.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import setuptools
+
+name = 'google-cloud-irm'
+description = 'Stackdriver Incident Response & Managment API client library'
+version = '0.1.0'
+release_status = 'Development Status :: 3 - Alpha'
+dependencies = [
+    'google-api-core[grpc] >= 1.6.0, < 2.0.0dev',
+    'google-cloud-monitoring',
```
Comment (busunkim96) [inline irm/setup.py]: IRM has a dependency on monitoring. How should the version be pinned?

Comment (tseaver) [reply irm/setup.py]: Ugh, `google-cloud-monitoring` is still in alpha status.  I would say use the current minor release as a minimum, and cap it at the next minor release, e.g.:

```python
    'google-cloud-monitoring >= 0.31.1, < 0.32dev'
```

Comment (tseaver) [reply irm/setup.py]: Also, note that you'll need to blacken first, which will morph all those lovely single quotes into doubles. :)

File: irm/google/cloud/irm_v1alpha2/gapic/transports/incident_service_grpc_transport.py [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+    google.cloud.irm.v1alpha2 IncidentService API.
+
+    The transport provides access to the raw gRPC stubs,
+    which can be used to take advantage of advanced
+    features of gRPC.
+    """
+
+    # The scopes needed to make gRPC calls to all of the methods defined
+    # in this service.
+    _OAUTH_SCOPES = ()
```
Comment (tseaver) [inline irm/google/cloud/irm_v1alpha2/gapic/transports/incident_service_grpc_transport.py]: @jdpedrie pointed out that the GAPIC config for IRM is missing the scoping information needed here.

File: docs/index.rst [hunk 3]
```diff
@@ -36,6 +36,7 @@
   Vision <vision/index>
   Video Intelligence <videointelligence/index>
   Stackdriver Error Reporting <error-reporting/index>
+  Stackdriver Incident Response & Managment <irm/index>
```
Comment (crwilcox) [inline docs/index.rst]:
```suggestion
  Stackdriver Incident Response & Management <irm/index>
```

File: irm/google/cloud/irm_v1alpha2/gapic/enums.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    class Type(enum.IntEnum):
+        """
+        Possible types of an artifact.
+
+        Attributes:
+          TYPE_UNSPECIFIED (int): External type is unspecified.
+          TYPE_URL (int): URL.
+          TYPE_ISSUE (int): An issue.
+          TYPE_SOURCE_CONTROL_CHANGE (int): A source control change.
+          TYPE_JIRA_ISSUE (int): A JIRA issue.
```
Comment (crwilcox) [inline irm/google/cloud/irm_v1alpha2/gapic/enums.py]: This seems odd to me. we have `issue` and `jira_issue`. What is `issue` representing. Is there a default issue db?

File: irm/noxfile.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        "black",
+        "--check",
+        "google",
+        "tests",
+        "docs",
+    )
+    session.run("flake8", "google", "tests")
+
+
+@nox.session(python="3.6")
```
Comment (crwilcox) [inline irm/noxfile.py]: It seems all of the other sessions are running the latest python. Is there a reason black isn't?
```suggestion
@nox.session(python="3.7")
```

**Comment by busunkim96** [comment]
Is it okay to merge this now and make follow-up PRs for other changes? @crwilcox

File: irm/google/cloud/irm_v1alpha2/gapic/enums.py (hunk 4 above)
Comment (tseaver) [reply irm/google/cloud/irm_v1alpha2/gapic/enums.py]: @busunkim96 The GAPIC config for IRM has been [updated to include scopes](https://github.com/googleapis/googleapis/commit/84f9bf2c274694a632b67ca0efcad20c56ead8cd).

File: irm/google/cloud/irm_v1alpha2/gapic/transports/incident_service_grpc_transport.py (hunk 2 above)
Comment (tseaver) [reply irm/google/cloud/irm_v1alpha2/gapic/transports/incident_service_grpc_transport.py]: @busunkim96 The GAPIC config for IRM has been [updated to include scopes](https://github.com/googleapis/googleapis/commit/84f9bf2c274694a632b67ca0efcad20c56ead8cd).

Comment (busunkim96) [reply irm/google/cloud/irm_v1alpha2/gapic/transports/incident_service_grpc_transport.py]: Thank you, I'll re-generate.

File: irm/noxfile.py (hunk 5 above)
Comment (tseaver) [reply irm/noxfile.py]: Hmm, looks like it is codegen which is generating the "3.6".

**Pull request merged**